### ISSUE TITLE
Memory leak in recv() callback on Win10

### ISSUE THREAD

**Issue by disazoz** [frida/frida-gum#421]
I'm sending data from python to js code, and looks like js GC do not free up that data after function exit. And there is no way to point GC to free it.

**python**
``` python
def on_message(message, data):
   script.post({'type': 'image_buf', 'payload': 'test'}, data=buf)
```
where **buf** is a large bytes array (around 2 MB, ie buf=bytes(some_data))

**js**
``` js
function test_cb(message, data) {
	var p = data.unwrap()
	Memory.copy(CameraGrayValueBuffer, p, 1167360);
}

var op = recv('input', test_cb);
send('ping');
op.wait();
```

But, if i add to js callback this:
``` js
function test_cb(message, data) {
	var p = data.unwrap()
	Memory.copy(CameraGrayValueBuffer, p, 1167360);
        var tmp = ptr(data)    // <======= added this
}
```
frida js will fire to me this error (with i can just ignore, and my js keep working)

    "type": "error",
    "description": "Error: expected a pointer",
    "stack": "Error: expected a pointer
    at frida/runtime/core.js:94
    at test_cb (/script1.js:215)
    at frida/runtime/message-dispatcher.js:54
    at s (frida/runtime/message-dispatcher.js:40)
    at forEach (native)
    at c (frida/runtime/message-dispatcher.js:31)
    at o (frida/runtime/message-dispatcher.js:25)",
    "fileName": "frida/runtime/core.js",
    "lineNumber": 94,
    "columnNumber": 1

But in this case **data** correctly freed up at function exit. Is there any ideas how can i fix this?

windows 10 x64, python 3.7, frida 12.8.20, runtime='duk', target is 64 bit mfc c++ windows gui exe

**Comment by oleavr** [comment]
@disazoz Thanks for reporting! Just added a test for this in fd9690e. It's currently failing w/ the Duktape runtime, and passing on the V8 runtime.

**Comment by oleavr** [comment]
Fix released in Frida 12.9.0.
